boxcli/shell_test.go

```diff
@@ -0,0 +1,286 @@
+// Copyright 2022 Jetpack Technologies Inc and contributors. All rights reserved.
+// Use of this source code is governed by the license in the LICENSE file.
+package boxcli
+
+import (
+	"bufio"
+	"errors"
+	"fmt"
+	"io"
+	"os"
+	"os/exec"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+)
+
+func testShellHello(t *testing.T, name string) {
+	// Skip this test if the required shell isn't installed, unless we're
+	// running in CI.
+	ci, _ := strconv.ParseBool(os.Getenv("CI"))
+	if _, err := exec.LookPath(name); err != nil && !ci {
+		t.Skipf("Skipping because %s isn't installed or in your PATH.", name)
+	}
+
+	sh := newShell(t, name)
+	sh.parentIO.write(t, "devbox init")
+	sh.parentIO.write(t, "devbox add hello")
+	sh.startDevboxShell(t)
+
+	sh.devboxIO.write(t, `echo "My name is: $0"`)
+	out := sh.devboxIO.read(t)
+	if !strings.HasSuffix(out, name) {
+		t.Errorf("Shell says its name is %q, but want it to contain %q.", out, name)
+	}
+
+	sh.devboxIO.write(t, "hello")
+	out = sh.devboxIO.read(t)
+	want := "Hello, world!"
+	if out != "Hello, world!" {
+		t.Errorf("Got hello command output %q, want %q.", out, want)
+	}
+}
+
+func TestShellHelloBash(t *testing.T) { testShellHello(t, "bash") }
+func TestShellHelloDash(t *testing.T) { testShellHello(t, "dash") }
+func TestShellHelloZsh(t *testing.T)  { testShellHello(t, "zsh") }
```

Comment:
If we wanted to be really exhaustive, this could be the list:
- Bash
- Cmd
- Dash
- Elvish
- Fish
- Ion
- Nushell
- PowerShell
- Tcsh
- Xonsh
- Zsh

This is based on the list that `starship` works with: https://starship.rs/guide/#%F0%9F%9A%80-installation

Reply:
Yeah, there are a bunch more we could support. It's a bit of work because we have to implement the detection and using a custom shellrc. I should probably add some documentation on which shells we officially test and support.